addons/microsoft_calendar/utils/microsoft_calendar.py

```diff
@@ -25,75 +23,159 @@ def wrapped(self, *args, **kwargs):
+        )
+
+        # and then, loop on other pages to get all the events
         events = data.get('value', [])
         next_page_token = data.get('@odata.nextLink')
         while next_page_token:
-            status, data, time = self.microsoft_service._do_request(next_page_token, {}, headers, preuri='', method='GET', timeout=timeout)
+            _, data, _ = self.microsoft_service._do_request(
+                next_page_token, {}, headers, preuri='', method='GET', timeout=timeout
+            )
             next_page_token = data.get('@odata.nextLink')
             events += data.get('value', [])
 
-        next_sync_token_url = data.get('@odata.deltaLink')
-        next_sync_token = urls.url_parse(next_sync_token_url).decode_query().get('$deltatoken', False)
+        token_url = data.get('@odata.deltaLink')
+        next_sync_token = urls.url_parse(token_url).decode_query().get('$deltatoken', False) if token_url else None
 
-        default_reminders = data.get('defaultReminders')
+        return events, next_sync_token
+
+    @requires_auth_token
+    def _get_events_delta(self, sync_token=None, token=None, timeout=TIMEOUT):
+        """
+        Get a set of events that have been added, deleted or updated in a time range.
+        See: https://docs.microsoft.com/en-us/graph/api/event-delta?view=graph-rest-1.0&tabs=http
+        """
+        url = "/v1.0/me/calendarView/delta"
+        params = {'$deltatoken': sync_token} if sync_token else None
 
-        return MicrosoftEvent(events), next_sync_token, default_reminders
+        try:
+            events, next_sync_token = self._get_events_from_paginated_url(
+                url, params=params, token=token, timeout=timeout)
+        except requests.HTTPError as e:
+            if e.response.status_code == 410 and 'fullSyncRequired' in str(e.response.content) and sync_token:
+                # retry with a full sync
+                return self._get_events_delta(token=token, timeout=timeout)
+            raise e
+
+        # event occurrences (from a recurrence) are retrieved separately to get all their info,
+        # # and mainly the iCalUId attribute which is not provided by the 'get_delta' api end point
+        events = list(filter(lambda e: e.get('type') != 'occurrence', events))
+
+        return MicrosoftEvent(events), next_sync_token
+
+    @requires_auth_token
+    def _get_occurrence_details(self, serieMasterId, token=None, timeout=TIMEOUT):
+        """
+        Get all occurrences details from a serie master.
+        See: https://docs.microsoft.com/en-us/graph/api/event-list-instances?view=graph-rest-1.0&tabs=http
+        """
+        url = f"/v1.0/me/events/{serieMasterId}/instances"
+
+        events, _ = self._get_events_from_paginated_url(url, token=token, timeout=timeout)
+        return MicrosoftEvent(events)
+
+    @requires_auth_token
+    def get_events(self, sync_token=None, token=None, timeout=TIMEOUT):
```

Comment:
Why does this take a timeout? Seems unused and is confusing as the method looks "public" (but is not because it's on the service, not on a model).

Reply:
The `timeout` parameter is used inside the method, but set everywhere to the `TIMEOUT` constant so yes, it is not really useful...
In master, we are working on a rework/refactor of this module so this point should be taken into account.